### PR TITLE
Replace HTTPlug factories by PSR-17

### PR DESCRIPTION
**Pull request by nicolas-grekas**
All interfaces of `php-http/message-factory` are now deprecated.
This PR provides a way to move to PSR-17 instead.

### CODE REVIEW

File: src/Http/Provider/AbstractHttpProvider.php [hunk 1]
```diff
@@ -33,18 +38,14 @@ abstract class AbstractHttpProvider extends AbstractProvider
...
+     * @var RequestFactoryInterface&StreamFactoryInterface
      */
     private $messageFactory;
 
-    /**
-     * @param ClientInterface     $client
-     * @param MessageFactory|null $factory
-     */
-    public function __construct(ClientInterface $client, MessageFactory $factory = null)
+    public function __construct(ClientInterface $client)
```
Comment (nicolas-grekas) [inline src/Http/Provider/AbstractHttpProvider.php:45]: the removed argument is never used: all implementations override the constructor without passing a second argument ever

**Comment by nicolas-grekas** [comment]
Failures will be fixed once 4.4 will be tagged.

**Comment by norkunas** [comment]
@jbelien could you review/merge this? :)

**Review by Nyholm — APPROVED** [review]
Thank you. 

If we do a new Major after this PR, then we can also remove the `getMessageFactory()` method instead of deprecate it.

**Comment by jbelien** [comment]
Thanks for the review @Nyholm
Good idea indeed to create a new major release so we can also remove `getMessageFactory()`. 👍 

@nicolas-grekas Could you fix the 8 failing tests ? I had a quick look and it seems related to `createRequest()` function.

**Review by jbelien — DISMISSED** [review]
Could you fix the 8 failing tests ? I had a quick look and it seems related to `createRequest()` function.

**Comment by nicolas-grekas** [comment]
Needs [src/Provider/AlgoliaPlaces/AlgoliaPlaces.php](https://github.com/geocoder-php/provider-integration-tests/pull/20) before I can move forward.

**Comment by jbelien** [comment]
> Needs [geocoder-php/provider-integration-tests#20](https://github.com/geocoder-php/provider-integration-tests/pull/20) before I can move forward.

Woops! Sorry, I've missed that one. Checking it now 👍

**Comment by ker0x** [comment]
Hello @nicolas-grekas @jbelien ! Is there any progress on this PR?

**Comment by jbelien** [comment]
> Hello @nicolas-grekas @jbelien ! Is there any progress on this PR?

I've just merged https://github.com/geocoder-php/provider-integration-tests/pull/20 and created a new release ; @nicolas-grekas can now update this PR. 👍

File: src/Http/Provider/AbstractHttpProvider.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Http/Provider/AbstractHttpProvider.php]: fixed

**Comment by nicolas-grekas** [comment]
PR rebased and ready. The two failures are related to the subtree splitting and need v4.6.0 of the common-http package to be released.

**Comment by Chris53897** [comment]
@jbelien Is there a planned release date for version 4.6 for https://github.com/geocoder-php/php-common-http

**Comment by Stadly** [comment]
Looking forward to this being released :)

**Comment by ruudk** [comment]
@Nyholm You are probably very busy, but do you know what is needed to move this PR forward? Thanks 💙

File: src/Http/composer.json [hunk 2]
```diff
@@ -15,12 +15,10 @@
     ],
     "require": {
         "php": "^8.0",
-        "php-http/client-implementation": "^1.0",
-        "php-http/discovery": "^1.6",
-        "php-http/httplug": "^1.0 || ^2.0",
-        "php-http/message-factory": "^1.0.2",
-        "psr/http-message": "^1.0 || ^2.0",
-        "psr/http-message-implementation": "^1.0",
+        "php": "^7.4 || ^8.0",
```
Comment (Nyholm) [inline src/Http/composer.json:18]: This looks like an issue while rebasing. Please remove this line and support php8 only.

**Review by Nyholm — APPROVED** [review]
Thank you. 

I think it all looks good. Just one minor

**Comment by jbelien** [comment]
> @jbelien I am happy to merge.

Thanks a lot @Nyholm for taking care of it! 😄 
Sorry last few months have been very busy.
I've dismissed my review so it's not blocking.

**Comment by Nyholm** [comment]
No worries. That is how it is sometimes. 

Let's move forward with this =)

**Comment by Nyholm** [comment]
Thank you for the PR and the reviews

**Pull request merged**